pkg/gateway/mirror/service/gateway_mock_test.go

```diff
@@ -26,11 +25,11 @@ type GatewayMock struct {
 	ForwardedContextFunc     func(ctx context.Context, podName string) context.Context
 	FromForwardedContextFunc func(ctx context.Context) string
 	BroadCastFunc            func(ctx context.Context,
-		f func(ctx context.Context, target string, vc vald.ClientWithMirror, copts ...grpc.CallOption) error) error
+		f func(ctx context.Context, target string, vc MirrorClient, copts ...grpc.CallOption) error) error
 	DoFunc func(ctx context.Context, target string,
-		f func(ctx context.Context, target string, vc vald.ClientWithMirror, copts ...grpc.CallOption) (interface{}, error)) (interface{}, error)
+		f func(ctx context.Context, target string, vc MirrorClient, copts ...grpc.CallOption) (interface{}, error)) (interface{}, error)
 	DoMultiFunc func(ctx context.Context, targets []string,
-		f func(ctx context.Context, target string, vc vald.ClientWithMirror, copts ...grpc.CallOption) error) error
+		f func(ctx context.Context, target string, vc MirrorClient, copts ...grpc.CallOption) error) error
 	GRPCClientFunc func() grpc.Client
 }
```

Comment:
Maybe `GatewayMock` should be moved to `internal/test/mock` as well?

Reply:
Thank you for your comment 🙏 
I replied in the following comment !!
https://github.com/vdaas/vald/pull/2366#discussion_r1495302645